prediction_market_agent/agents/microchain_agent/market_functions.py

```diff
@@ -312,6 +317,35 @@ def __call__(self) -> list[str]:
         return [str(position) for position in positions]
 
 
+class GetResolvedBetsWithOutcomes(MarketFunction):
+    def __init__(self, market_type: MarketType) -> None:
+        self.user_address = APIKeys().bet_from_address
+        super().__init__(market_type=market_type)
+
+    @property
+    def description(self) -> str:
+        return (
+            "Use this function to fetch the outcomes of previous bets you have placed."
+        )
+
+    @property
+    def example_args(self) -> list[str]:
+        return []
+
+    def __call__(self) -> list[dict[str, t.Any]]:
+        subgraph_handler = OmenSubgraphHandler()
+        # We look back a standard interval as a rule-of-thumb for now.
+        start_time = utcnow() - timedelta(days=7)
+        bets = subgraph_handler.get_resolved_bets_with_valid_answer(
+            better_address=self.user_address,
+            start_time=start_time,
+            end_time=None,
+            market_id=None,
+        )
+        generic_bets = [b.to_generic_resolved_bet() for b in bets]
+        return [b.model_dump() for b in generic_bets]
```

Comment:
I tried selecting just some of the keys but couldn't decide what was most relevant, hence dumped them all - do you have any suggestions regarding this, or just having a more readable __str__ already solves this?

Reply:
Yeah I'd say we want two things:
- add a __str__ to `TokenAmount`
- replace `market_question` property of `Bet` with `market_id` (or just add it as a new field?)

and then it would just be something like `f"{created_time}: Bet {amount} on outcome get_outcome_str(outcome) {market_id}. Correct/Incorrect, so won/lost {profit}"`

Reply:
Implemented in PMAT (see for ex https://github.com/gnosis/prediction-market-agent-tooling/blob/4cb46ebe1f9e0ab32afb852dc84ce39b40940467/prediction_market_agent_tooling/markets/data_models.py#L56-L57)